src/ssl/support.h

```diff
@@ -65,6 +65,10 @@ class MemMap;
 
 namespace Ssl
 {
+
+/// callback for receiving password to access password secured PEM files
+int AskPasswordCb(char *buf, int size, int rwflag, void *userdata);
```

Comment:
This I am a bit reluctant to add just yet. IMO if an admin wants a level of security where PEM file access has to be secured that tight the GnuTLS implementation is not yet ready for their use. It is "good enough" for a limited range of very simple configurations and still quite experimental even there.

FWIW the GnuTLS requires we send a password string at the same time we pass the filename(s) for cert and key file. So to implement this we need to rearrange the config dependencies so that sslpassword_program is available to be run before any PEM files are loaded. That all needs to be done while ensuring that OpenSSL's very different behaviour does not break, and also that we are not holding onto a password in memory too long.

For now we do not deliver the password string at all. So the loading of password protected PEM files should fail cleanly and explicitly with an error mentioning that file protection.

Reply:
I was asking for a password-asking API fix. You are describing specifics/limitations of GnuTLS-related code. I believe the fixed API can serve the current code, at least for now. I do not insist on this change.

Reply:
Ah, sorry I misunderstood that as a request to implement the assk-password feature.  Will add the XXX.